branching_xblock/branching_xblock.py

```diff
@@ -249,107 +258,107 @@ def select_choice(self, data, suffix=''):
             if self.enable_scoring:
                 self.score = self.max_score
                 self.publish_grade()
+            # You should also add `has_custom_completion = True` to the XBlock to
+            # make sure this works as expected.
             self.runtime.publish(self, "completion", {"completion": 1.0})
-    
-        return {"success": True}
-    
+
+        return {"success": True, **self._get_state()}
+
     @XBlock.json_handler
     def undo_choice(self, data, suffix=''):
         if not self.enable_undo or not self.history:
             return {"success": False, "error": "Undo not allowed"}
-        
+
         prev_node_id = self.history.pop()
         self.current_node_id = prev_node_id
 
         if self.has_completed and self.enable_scoring:
             self.score = 0.0
             self.publish_grade()
-        
+
         self.has_completed = False
-        return {"success": True}
-    
-    @XBlock.json_handler
-    def save_settings(self, data, suffix=''):
-        self.enable_undo = data.get("enable_undo", False)
-        self.enable_scoring = data.get("enable_scoring", False)
-        self.max_score = float(data.get("max_score", 100.0))
-        return {"success": True}
-    
+        return {"success": True, **self._get_state()}
+
     @XBlock.json_handler
-    def add_node(self, data, suffix=''):
-        new_id = f"node-{uuid.uuid4().hex[:6]}"
-        while any(n["id"] == new_id for n in self.scenario_data["nodes"]):
-            new_id = f"node-{uuid.uuid4().hex[:6]}"
-        new_node = {
-            "id": new_id,
-            "type": "normal",
-            "content": "New Node",
-            "choices": []
+    def studio_submit(self, data, suffix=''):
```

Comment:
This doesn't seem to validate if there is a choice text set. by default it will be empty which shows an empty button.

Reply:
Thanks for catching that. Fixed it.